Stripe/STPAddressViewModel.m

```diff
@@ -167,6 +170,60 @@ - (void)addressFieldTableViewCellDidReturn:(STPAddressFieldTableViewCell *)cell
     [[self cellAfterCell:cell] becomeFirstResponder];
 }
 
+- (void)addressFieldTableViewCellDidEndEditing:(STPAddressFieldTableViewCell *)cell {
+    if (cell.type == STPAddressFieldTypeZip) {
+        [self updateCityAndStateFromZipCodeCell:cell];
+    }
+}
+
+- (void)updateCityAndStateFromZipCodeCell:(STPAddressFieldTableViewCell *)zipCell {
+
+    if (!zipCell.textField.validText
+        || ![_addressFieldTableViewCountryCode isEqualToString:@"US"]) {
+        return;
+    }
+
+    NSString *zipCode = zipCell.contents;
+    STPAddressFieldTableViewCell *cityCell = nil;
+    STPAddressFieldTableViewCell *stateCell = nil;
+    for (STPAddressFieldTableViewCell *cell in self.addressCells) {
+        if (cell.type == STPAddressFieldTypeCity) {
+            cityCell = cell;
+        }
+        else if (cell.type == STPAddressFieldTypeState) {
+            stateCell = cell;
+        }
+    }
+
+    if ((cityCell == nil && stateCell == nil)
+        || (cityCell.contents.length > 0 || stateCell.contents.length > 0)) {
+        // Don't auto fill if either have text already
+        // Or if neither are non-nil
+        return;
+    }
+    else {
+
+        CLGeocoder *geocoder = [CLGeocoder new];
+        NSString *zipKey = (NSString *) kABPersonAddressZIPKey;
+        NSString *countryCodeKey = (NSString *) kABPersonAddressCountryCodeKey;
+
+        if (zipKey && countryCodeKey) {
```

Comment:
imo there's no harm in over-checking/being paranoid to prevent dict/array crashes – can we add just to be safe?
 could also nil-coalesce in the dictionary, e.g. `zipKey : zipCode ?: @""`.